Nodejs/Product/Npm/SPI/NpmController.cs

```diff
@@ -100,18 +95,19 @@ internal class NpmController : AbstractNpmLogSource, INpmController {
         }
 
         public void Refresh() {
-            RefreshAsync().ContinueWith(t => {
-                var ex = t.Exception;
+            try {
+                RefreshImplementation();
+            } catch (Exception ex) {
                 if (ex != null) {
                     OnOutputLogged(ex.ToString());
 #if DEBUG
                     Debug.Fail(ex.ToString());
 #endif
                 }
-            });
+            }
         }
 
-        public async Task RefreshAsync() {
+        private void RefreshImplementation() {
```

Comment:
Ahh, I see the reason behind changing the behavior. I would prefer not to take this additional refactoring for this particular change, but rather in a separate PR